internal/buildjob/buildjob.go

```diff
@@ -61,17 +61,19 @@ func New(cfg Config) (*Job, error) {
 
 	var cleanupSteps []func()
 
+	var publisher message.Publisher
 	// setup message publisher
-	var producer message.Producer
+	log.Info("Set up message publisher if options are configured")
 	if cfg.BrokerOpts != nil {
 		log.Info("Initializing status update message publisher")
 
-		if producer, err = message.NewProducer(cfg.BrokerOpts); err != nil {
-			return nil, err
+		if publisher = message.NewPublisher(cfg.BrokerOpts, log); publisher == nil {
```

Comment:
error checking handled a bit differently from before, in that the Publisher constructor doesn't return an error as a default case